src/libsyntax/feature_gate.rs

```diff
@@ -385,6 +386,13 @@ impl<'a> Context<'a> {
                 return;
             }
         }
+        for &(ref n, ref ty) in self.plugin_attributes.iter() {
+            if &*n == name {
+                // Plugins can't gate attributes, so we don't check for it
```

Comment:
This code does feature gating of specific attrs; even if custom_attribute is on, attributes like `rustc_on_unimplemented` can't be used until the specific feature is turned on. `AttributeType` has a `Gated` variant that we check for in the code above, but we don't do that here since `plugin_attributes` cannot contain `Gated` types.